service/src/main/java/com/epam/esm/service/GiftCertificateService.java

```diff
@@ -160,14 +164,19 @@ public GiftCertificateDto update(GiftCertificateDto certificateDto) {
 
         LocalDateTime lastUpdateDate = LocalDateTime.now(UTC);
         certificate.setLastUpdateDate(lastUpdateDate);
-        certificateRepository.update(certificate);
+        List<String> tagNames = certificateDto.getTags();
 
         // we do not update tags if it is not specified in request (i.e. tagNames == null)
         if (tagNames != null) {
-            processTags(certificate.getId(), tagNames);
+            List<Tag> tags = certificateDto.getTags()
+                    .stream()
+                    .map(this::processTag)
+                    .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Probably we can move this code (lines 119 - 122 and 171 - 174) to a method? To avoid duplication

Reply:
Fixed.